tools/pipelines/templates/build-npm-package.yml

```diff
@@ -415,12 +438,25 @@ stages:
               artifactName: '_api-extractor-temp'
               publishLocation: 'pipeline'
 
+        - ${{ if eq(parameters.packageManager, 'pnpm') }}:
+          # Reset the pnpm-lock.yaml file since it's been modified by the versioning. But for dependency caching we want
+          # the cache key (which is based on the contents of the lockfile) to be the unmodified file. So we reset the
+          # lockfile as the last step so that when the dependency cache is uploaded, the cache key matches what it was
+          # at the beginning of the CI job.
```

Comment:
👍 love the clarification. I wouldn't have guessed that the upload of the cache (and computation of the file hash) occurs at the end, so this step would feel redundant without the comment.

Reply:
I think the store has to be uploaded at the end since we run `pnpm i` as part of the build, which is what would populate the store initially.

(for posterity only) The computation of the hash is done both at the beginning -- to determine if there's a cache hit -- and at the end to re-verify the cache key before uploading (smart - I would not have thought to do this). Before I added this step, I would see a warning saying that the file hash had changed, which effectively made the cache useless since each uploaded cache would have a cache key for a _single build._